### PR TITLE
REF: simplify PeriodIndex.get_loc

### PR DESCRIPTION
**Pull request by jbrockmendel**
This makes PeriodEngine follow patterns in DatetimeEngine, so soon we can have it subclass DatetimeEngine instead of Int64Engine, which will lead to more simplifications.

This also puts us within striking distance of sharing get_loc code

### CODE REVIEW

File: pandas/core/indexes/period.py [hunk 1]
```diff
@@ -545,20 +550,12 @@ def get_loc(self, key, method=None, tolerance=None):
             key = Period(key, freq=self.freq)
         except ValueError:
             # we cannot construct the Period
-            raise KeyError(key)
+            raise KeyError(orig_key)
 
-        ordinal = key.ordinal if key is not NaT else key.value
         try:
-            return self._engine.get_loc(ordinal)
+            return Index.get_loc(self, key, method, tolerance)
```
Comment (jreback) [inline pandas/core/indexes/period.py]: I wonder if this changes perf? the reason is that we have already created an _int64index, but now we are creating something new?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/period.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/period.py]: > but now we are creating something new?

I don't think we're creating a new object here

Comment (jreback) [reply pandas/core/indexes/period.py]: the engine itself is (maybe) being created again

Comment (jbrockmendel) [reply pandas/core/indexes/period.py]: i guess on the first all, but on the flip side the _int64index may not have been created yert

File: pandas/tests/indexing/test_indexing.py [hunk 2]
```diff
@@ -153,7 +153,6 @@ def test_setitem_ndarray_3d(self, index, obj, idxr, idxr_id):
                     "floating",
                     "string",
                     "datetime64",
-                    "period",
```
Comment (jreback) [inline pandas/tests/indexing/test_indexing.py]: what changed here?

Comment (jbrockmendel) [reply pandas/tests/indexing/test_indexing.py]: on the `Series.__setitem__` case it now (correctly, i think) raises, as it is trying to instantiate an index with 3D data

Comment (jbrockmendel) [reply pandas/tests/indexing/test_indexing.py]: this test is a mess, is hiding the fact that all of the cases in this block _should_ raise but currently dont  (xref #31813)

**Comment by jreback** [comment]
can you rebase

**Comment by jbrockmendel** [comment]
rebased+green

**Review by jreback — APPROVED** [review]

**Pull request merged**